go/vt/vttablet/onlineddl/executor.go

```diff
@@ -2017,6 +2017,10 @@ func (e *Executor) evaluateDeclarativeDiff(ctx context.Context, onlineDDL *schem
 // getCompletedMigrationByContextAndSQL chceks if there exists a completed migration with exact same
 // context and SQL as given migration. If so, it returns its UUID.
 func (e *Executor) getCompletedMigrationByContextAndSQL(ctx context.Context, onlineDDL *schema.OnlineDDL) (completedUUID string, err error) {
+	if onlineDDL.RequestContext == "" {
+		// only applies to migrations with an explicit context
+		return "", nil
+	}
```

Comment:
This is actually unrelated to the essence of the PR but something I cleaned up while I was here